src/regexp.c

```diff
@@ -2487,6 +2577,23 @@ reg_getline_submatch(linenr_T lnum)
     return s;
 }
 
+    static colnr_T
+reg_getline_submatchlen(linenr_T lnum)
+{
+    colnr_T len;
+    linenr_T save_first = rex.reg_firstlnum;
+    linenr_T save_max = rex.reg_maxline;
+
+    rex.reg_firstlnum = rsm.sm_firstlnum;
+    rex.reg_maxline = rsm.sm_maxline;
+
+    len = reg_getlinelen(lnum);
+
+    rex.reg_firstlnum = save_first;
+    rex.reg_maxline = save_max;
+    return len;
+}
```

Comment:
This duplicates code with the function above `reg_getline_submatch()`. Can we refactor both into a a common function?